sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -125,7 +127,7 @@ async def serialize_property(value: 'Input[Any]',
             remote_asset = cast('RemoteAsset', value)
             obj["uri"] = await serialize_property(remote_asset.uri, deps, input_transformer)
         else:
-            raise AssertionError(f"unknown asset type: {value}")
+            raise AssertionError(f"unknown asset type: {value!r}")
```

Comment:
Otherwise mypy errors:

```
error: On Python 3 '{}'.format(b'abc') produces "b'abc'"; use !r if this is a desired behavior
```

due to the `not isinstance(value, bytes)` above.